src/main/java/org/snomed/snowstorm/core/data/services/TraceabilityLogService.java

```diff
@@ -65,18 +75,83 @@ public TraceabilityLogService() {
 	}
 
 	@Override
-	public void preCommitCompletion(Commit commit) throws IllegalStateException {
-		if (commit.getCommitType() != CONTENT) {// Only use for rebase / promotion
-			logActivity(SecurityUtil.getUsername(), commit, new PersistedComponents());
+	public void preCommitCompletion(final Commit commit) throws IllegalStateException {
+		// Most commits don't use this because ConceptService calls logActivity directly.
+
+		PersistedComponents persistedComponents = null;
+		String importType = null;
+		final Map<String, String> metadata = commit.getBranch().getMetadata();
+		if (metadata != null && metadata.containsKey(ImportService.IMPORT_TYPE_KEY)) {
+			importType = metadata.get(ImportService.IMPORT_TYPE_KEY);
+			if (commit.getCommitType() == CONTENT && RF2Type.DELTA.getName().equals(importType)) {
+				// RF2 Delta import
+				persistedComponents = persistedComponentLoaderChainService.load(commit);
+			}
+		} else {
+			// Rebase or Promotion
+			// Log an empty-change activity.
+			persistedComponents = new PersistedComponents();
 		}
+
+		logActivity(SecurityUtil.getUsername(), commit, persistedComponents, false, importType);
 	}
 
-	void logActivity(String userId, Commit commit, PersistedComponents batchSavePersistedComponents) {
+	/**
+	 * Logs the activity that is associated to the commit. It should be noted
+	 * that some services call this method explicitly so that means they will
+	 * have the states in memory for checking whether the component is changed
+	 * or deleted, both of these fields are classified as {@code Transient} so
+	 * the states will not be persisted when retrieved from the data store.
+	 * In the case where this method is called implicitly through the
+	 * {@link CommitListener}, by performing a {@link RF2Type#DELTA} import,
+	 * it will load the components from the data store so the {@code changed}
+	 * and {@code deleted} states will not be persisted which means that the
+	 * <code>useChangeFlag == false</code> so that other checks are ignored when
+	 * doing the comparison between what has changed. It's worth noting when doing
+	 * a {@link RF2Type#DELTA} import, it will show all the changes inside the
+	 * traceability log.
```

Comment:
It would be better to keep comments as brief and to the point as possible. Very long comments may be ignored and are likely to become out of date, especially when some of the information in them is about the workings of another class. The key information here is: "This method may be called by a service class as an optimisation. In this case the transient change and delete flags may be populated."

Reply:
Fixed